Hot reloading works just once

I try to make an admin action for hot reloading.

The `probe` action works ok (even if I call it many times):
```
$ make probe
cartridge admin --run-dir tmp/run --name meta probe --uri localhost:3301
Probe "localhost:3301": OK
$ make probe
cartridge admin --run-dir tmp/run --name meta probe --uri localhost:3301
Probe "localhost:3301": OK
```

But my custom `reload` action works once:
```
meta.router | 2023-09-11 10:12:48.485 [58114] main/148/vshard.discovery._static_router/vshard.router I> Updated replicaset(uuid="29690241-8013-4c2e-9bbb-39a894f3ab97", master=localhost:3304(admin@localhost:3304)) buckets: was 14000, became 15000
meta.router | 2023-09-11 10:12:48.485 [58114] main/148/vshard.discovery._static_router/vshard.router I> Updated replicaset(uuid="79089f8b-e99a-443d-9942-00d7443ba80d", master=localhost:3302(admin@localhost:3302)) buckets: was 14000, became 15000
meta.router | 2023-09-11 10:12:48.485 [58114] main/148/vshard.discovery._static_router/vshard.router I> Discovery enters idle mode, all buckets are known. Discovery works with 10 seconds interval now
meta.router | 2023-09-11 10:13:09.142 [58114] main/178/console/unix/:/cartridge.roles roles.lua:575 W> Reloading roles ...
...
``` 

But when I call it again it crashed:
```
$ make reload
cartridge admin --run-dir tmp/run --name meta reload
   ⨯ Failed to parse function call args: Failed to get function "reload" signature: [string "local function func(...) local res, err = __c..."]:1: variable '__cartridge_admin_help' is not declared. Please update cartridge-cli-extensions module or file an issue https://github.com/tarantool/cartridge-cli-extensions/issues/new
make: *** [reload] Error 1
```
and after that `probe`
also crashed:
```
$ make probe
cartridge admin --run-dir tmp/run --name meta probe --uri localhost:3301
   ⨯ Failed to parse function call args: Failed to get function "probe" signature: [string "local function func(...) local res, err = __c..."]:1: variable '__cartridge_admin_help' is not declared. Please update cartridge-cli-extensions module or file an issue https://github.com/tarantool/cartridge-cli-extensions/issues/new
make: *** [probe] Error 1
```

```lua
-- app/admin.lua
local cli_admin = require('cartridge-cli-extensions.admin')
local cartridge = require('cartridge')

local function probe()
  local usage = 'Probe instance'
  local args = {
    uri = {
      type  = 'string',
      usage = 'Instance URI',
    },
  }
  local call = function(opts)
    opts = opts or {}

    if opts.uri == nil then
      return nil, "Please, pass instance URI via --uri flag"
    end

    local ok, err = cartridge.admin_probe_server(opts.uri)

    if not ok then
      return nil, err.err
    end

    return {
      string.format('Probe %q: OK', opts.uri),
    }
  end

  local ok, err = cli_admin.register('probe', usage, args, call)
  assert(ok, err)
end

local function reload()
  local usage = 'Hot reload Cartridge roles'
  local args = {}
  local call = function(opts)
    local cartridge_roles = require('cartridge.roles')
    local ok, err = cartridge_roles.reload()

    if not ok then
      return nil, err.err
    end

    return {
      "Reload: Done",
    }
  end

  local ok, err = cli_admin.register('reload', usage, args, call)
  assert(ok, err)
end

-- register admin function probe to use it with "cartridge admin"
local function init()
  cli_admin.init()
  probe()
  reload()
end

return { init = init }
```

**PS**: 
Are there any right/standard method for hot reloading?
Perfectly would be with source code watch option (like `nodemon` in nodejs world).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Changed
+
+- Add global variables to Cartridge Hot Reload whitelist (#14).
+
 ## [1.1.1] - 2020-03-24
 
 ### Fixed

--- a/cartridge-cli-extensions/admin.lua
+++ b/cartridge-cli-extensions/admin.lua
@@ -164,6 +164,15 @@ function admin.init()
     rawset(_G, '__cartridge_admin_list', admin_list)
     rawset(_G, '__cartridge_admin_help', admin_help)
     rawset(_G, '__cartridge_admin_call', admin_call)
+
+    local ok, hotreload = pcall(require, 'cartridge.hotreload')
+    if ok then
+        hotreload.whitelist_globals({
+            '__cartridge_admin_list',
+            '__cartridge_admin_help',
+            '__cartridge_admin_call',
+        })
+    end
 end
 
 return admin

--- a/doc/admin.md
+++ b/doc/admin.md
@@ -51,6 +51,10 @@ Removes `func_name` function if it is registered.
 
 Always returns `true`.
 
+### Hot reload
+
+Cartridge Admin functions don't support Cartridge Hot Reload.
+Restart instances manually to add new admin functions.
 ## Example
 
 Describe admin function in application:

